### ISSUE TITLE
False negatives in no-control-regex

### ISSUE THREAD

**Issue by fasttime** [eslint/eslint#8397]
**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **Node Version:** 7.7.1
* **npm Version:** 4.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```
{ "parserOptions": { "ecmaVersion": 6 } }
```

**What did you do? Please include the actual source code causing the issue.**

```js
/*eslint no-control-regex: "error"*/

/\x0a/;   // error
/\u000a/; // no error
/\n/;     // no error
/\12/;    // no error
/\cJ/;    // no error
```

**What did you expect to happen?**

All the expressions above match the same character, so I would be expecting the rule to report errors for all of them, or for none.

**What actually happened? Please include the actual, raw output from ESLint.**

Only the sequence `\x0a` reports the error "Unexpected control character(s) in regular expression: \x0a.". 

Actually I'm not even sure if the line feed character was intended to trigger an error (`RegExp('\x0a')` doesn't, which could be another issue), but the issue with regexp literals applies to other characters, too, including `"\x1f"` quoted in the rule example.

**Comment by not-an-aardvark** [comment]
Related: https://github.com/eslint/eslint/issues/7476

**Comment by fasttime** [comment]
@not-an-aardvark Ah thanks, I missed that discussion. Since there is apparently no consensus about what the rule should actually do I'm going to change my mind and suggest to make the rule configurable as for what classes of escape sequences should be forbidden. It still feels broken as it's standing now.

**Comment by platinumazure** [comment]
Looks like this has stalled out. @not-an-aardvark What do you recommend for next steps?

**Comment by not-an-aardvark** [comment]
I think the best option for now would be to improve the rule's documentation to clarify that the rule only checks control codes.

**Comment by not-an-aardvark** [comment]
Closing because I think the current behavior (only reporting unicode escapes like `\uXXXX`) is working as intended.
